specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/StartStopManagedInstanceSchedules.json

```diff
@@ -0,0 +1,389 @@
+        "stopDay",
+        "stopTime"
+      ],
+      "type": "object",
+      "properties": {
+        "startDay": {
+          "description": "Start day.",
+          "enum": [
+            "Sunday",
+            "Monday",
+            "Tuesday",
+            "Wednesday",
+            "Thursday",
+            "Friday",
+            "Saturday"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "DayOfWeek",
+            "modelAsString": true
+          }
+        },
+        "startTime": {
+          "description": "Start time.",
+          "type": "string"
+        },
+        "stopDay": {
+          "description": "Stop day.",
+          "enum": [
+            "Sunday",
+            "Monday",
+            "Tuesday",
+            "Wednesday",
+            "Thursday",
+            "Friday",
+            "Saturday"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "DayOfWeek",
+            "modelAsString": true
+          }
+        },
+        "stopTime": {
+          "description": "Stop time.",
+          "type": "string"
+        }
+      }
+    },
+    "StartStopManagedInstanceSchedule": {
+      "description": "Managed instance's Start/Stop schedule.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "systemData": {
+          "$ref": "../../../common/v1/types.json#/definitions/systemData",
+          "description": "System data of the scheduled resource.",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/StartStopManagedInstanceScheduleProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "StartStopManagedInstanceScheduleListResult": {
+      "description": "Managed instance's Start/Stop schedule list result.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StartStopManagedInstanceSchedule"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "StartStopManagedInstanceScheduleProperties": {
+      "description": "Properties of managed instance's Start/Stop schedule.",
+      "required": [
+        "scheduleList"
+      ],
+      "type": "object",
+      "properties": {
+        "description": {
+          "description": "The description of the schedule.",
+          "default": "",
+          "type": "string"
+        },
+        "timeZoneId": {
+          "description": "The time zone of the schedule.",
+          "default": "UTC",
+          "type": "string"
```

Comment:
Do you support all timezones ? How does the client know what format to pass these in ?

Reply:
TimezoneId is already part of the SQL configuration during instance/sql server creation and same approach is applied for start/stop schedule API. So we are just re-using the existing approach.

It can be found as part of the documentation https://learn.microsoft.com/en-us/rest/api/sql/2022-05-01-preview/managed-instances/create-or-update?tabs=HTTP

**Per timezoneID description:**
properties.timezoneId
string
Id of the timezone. Allowed values are timezones supported by Windows. Windows keeps details on supported timezones, including the id, in registry under KEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Time Zones. You can get those registry values via SQL Server by querying SELECT name AS timezone_id FROM sys.time_zone_info. List of Ids can also be obtained by executing [System.TimeZoneInfo]::GetSystemTimeZones() in PowerShell. An example of valid timezone id is "Pacific Standard Time" or "W. Europe Standard Time".